### PR TITLE
scsynth: refactor SC_WorldOptions.h for readability

### PR DESCRIPTION
**Pull request by snappizz**
cleaning up some scsynth code for fun. with c++11 (currently required to build anyway) we have support for structs with default arguments, so the definition of default arguments in `WorldOptions` can be expressed much more readably.

### CODE REVIEW

File: include/server/SC_WorldOptions.h [hunk 1]
```diff
@@ -29,74 +29,62 @@
 
 typedef int (*PrintFunc)(const char *format, va_list ap);
 
-struct WorldOptions
+const struct WorldOptions
```
Comment (brianlheim) [inline include/server/SC_WorldOptions.h]: this is a really long struct definition. IMO clearer if you define the struct type here and then define the const instance below it.

File: include/server/SC_WorldOptions.h [hunk 2]
```diff
@@ -29,74 +29,62 @@
...
+	uint32 mMaxWireBufs = 64;
+	uint32 mNumAudioBusChannels = 1024;
+	uint32 mNumInputBusChannels = 8;
+	uint32 mNumOutputBusChannels = 8;
+	uint32 mNumControlBusChannels = 16384;
+	uint32 mBufLength = 64;
+	uint32 mRealTimeMemorySize = 8192; // in kilobytes
+
+	int mNumSharedControls = 0;
+	float *mSharedControls = 0;
```
Comment (brianlheim) [inline include/server/SC_WorldOptions.h]: -> nullptr

File: include/server/SC_WorldOptions.h [hunk 3]
```diff
@@ -29,74 +29,62 @@
...
-	const char *mNonRealTimeOutputHeaderFormat;
-	const char *mNonRealTimeOutputSampleFormat;
-
-	uint32 mPreferredSampleRate;
-	uint32 mNumRGens;
+#if defined(_WIN32)
+#	define DEFAULT_PREFERRED_SAMPLE_RATE 44100
+#else
+#	define DEFAULT_PREFERRED_SAMPLE_RATE 0
+#endif
```
Comment (brianlheim) [inline include/server/SC_WorldOptions.h]: this introduces a macro unnecessarily for all clients who include this header, plus doing it in the middle of the struct definition is icky. two options would be to do it without the macro name and just wrap the line below in #ifdef (which is OK by me) or to move this up above the struct definition and then #undef the macro after it's used. I think the first option is more in line with the spirit of what you're doing. I like that the definition is close to where it's used, just think that you don't need to go to all this trouble

**Comment by brianlheim** [comment]
This also fails CI because (I didn't know this before now) in clang pre-3.9 you need to provide a default constructor.

Example: https://godbolt.org/g/pqCi11

File: include/server/SC_WorldOptions.h [hunk 4]
```diff
@@ -29,74 +29,62 @@
...
-	uint32 mMaxNodes;
-	uint32 mMaxGraphDefs;
-	uint32 mMaxWireBufs;
-	uint32 mNumAudioBusChannels;
-	uint32 mNumInputBusChannels;
-	uint32 mNumOutputBusChannels;
-	uint32 mNumControlBusChannels;
-	uint32 mBufLength;
-	uint32 mRealTimeMemorySize;
+	const char* mPassword = nullptr;
```
Comment (brianlheim) [inline include/server/SC_WorldOptions.h]: so, just add a `WorldOptions(void) { }` here.

Comment (brianlheim) [reply include/server/SC_WorldOptions.h]: oh oh or `WorldOptions() = default;`

**Comment by snappizz** [comment]
thanks for the C++ help! reviewers -- this should be good to go, but please double check that i copied all the parameters correctly.

**Comment by brianlheim** [comment]
You could probably remove kDefaultWorldOptions from the codebase entirely. At a glance it seems there are only a few spots where it's used, and all of them could just rely on the default constructor.

File: server/scsynth/iPhone/iSCSynthController.mm [hunk 5]
```diff
@@ -36,7 +36,6 @@ - (id) init
 {
 	if (!theController && (self=[super init]))
 	{
-		options = kDefaultWorldOptions;
```
Comment (brianlheim) [inline server/scsynth/iPhone/iSCSynthController.mm]: where is this thing defined?

**Comment by vivid-synth** [comment]
Out of curiosity, when did we start depending on c++11?

**Comment by brianlheim** [comment]
@vivid-synth https://github.com/supercollider/supercollider/commit/736145e5caaf347b0de6c24bd23b2f9b8082e6bf

**Pull request merged**